portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,36 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+const addComments =
+    async () => {
+  try {
+    const commentTemplate = document.getElementById('comment-template');
+    const commentsContainer = document.getElementById('comments-container');
+    const response = await fetch('/data');
+    if (response.status < 200 ||
+        response.status > 299) {  // HTTP error codes in the 200 range are okay.
```

Comment:
This can be improved.

```
if (!response.ok()) {
  // throw the error.
}
```

In general, when you throw errors, you should make them as descriptive as possible so that they're easier to debug. In this case, you're only throwing a status code, but no accompanying error message. The best thing to do here would be to encapsulate the error in a new object with a message about what broke, the status code and error message.

```
throw new Error("Error occurred while fetching comment data.", response.status, response.statusText);
```